flujo/domain/dsl/pipeline_step_validations.py

```
@@ -153,27 +165,36 @@ def _validate_pipeline(
             if _ConditionalStep is not None and isinstance(step, _ConditionalStep):
-                branch_outputs: set[str] = set()
+                branch_paths_sets: list[set[str]] = []
                 branches = getattr(step, "branches", {}) or {}
                 for branch in branches.values():
                     try:
                         child_paths = _validate_pipeline(
                             branch, set(available_roots), set(produced_paths), None, None
                         )
-                        branch_outputs.update(child_paths)
+                        branch_paths_sets.append(set(child_paths))
                     except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)